x-pack/plugins/security_solution/public/common/lib/triggers_actions_ui/register_alerts_table_configuration.tsx

```diff
@@ -6,38 +6,100 @@
  */
 
 import type { Storage } from '@kbn/kibana-utils-plugin/public';
-import type {
-  AlertsTableConfigurationRegistryContract,
-  GetRenderCellValue,
-} from '@kbn/triggers-actions-ui-plugin/public';
+import type { AlertsTableConfigurationRegistryContract } from '@kbn/triggers-actions-ui-plugin/public';
 
-import { APP_ID, CASES_FEATURE_ID } from '../../../../common/constants';
+import type { AlertsTableConfigurationRegistry } from '@kbn/triggers-actions-ui-plugin/public/types';
+import { getUseTriggersActionsFieldBrowserOptions } from '../../../detections/hooks/trigger_actions_alert_table/use_trigger_actions_browser_fields_options';
+import { getUseCellActionsHook } from '../../../detections/hooks/trigger_actions_alert_table/use_cell_actions';
+import { getBulkActionHook } from '../../../detections/hooks/trigger_actions_alert_table/use_bulk_actions';
+import { getUseActionColumnHook } from '../../../detections/hooks/trigger_actions_alert_table/use_actions_column';
+import { getPersistentControlsHook } from '../../../detections/hooks/trigger_actions_alert_table/use_persistent_controls';
+import {
+  ALERTS_TABLE_REGISTRY_CONFIG_IDS,
+  APP_ID,
+  CASES_FEATURE_ID,
+} from '../../../../common/constants';
 import { getDataTablesInStorageByIds } from '../../../timelines/containers/local_storage';
 import { TableId } from '../../../../common/types';
 import { getColumns } from '../../../detections/configurations/security_solution_detections';
-import { useRenderCellValue } from '../../../detections/configurations/security_solution_detections/render_cell_value';
+import { getRenderCellValueHook } from '../../../detections/configurations/security_solution_detections/render_cell_value';
 import { useToGetInternalFlyout } from '../../../timelines/components/side_panel/event_details/flyout';
+import { SourcererScopeName } from '../../store/sourcerer/model';
 
 const registerAlertsTableConfiguration = (
   registry: AlertsTableConfigurationRegistryContract,
   storage: Storage
 ) => {
-  if (registry.has(APP_ID)) {
-    return;
-  }
   const dataTableStorage = getDataTablesInStorageByIds(storage, [TableId.alertsOnAlertsPage]);
   const columnsFormStorage = dataTableStorage?.[TableId.alertsOnAlertsPage]?.columns ?? [];
   const alertColumns = columnsFormStorage.length ? columnsFormStorage : getColumns();
 
+  const useInternalFlyout = () => {
+    const { header, body, footer } = useToGetInternalFlyout();
+    return { header, body, footer };
+  };
+
+  const renderCellValueHookAlertPage = getRenderCellValueHook({
+    scopeId: SourcererScopeName.detections,
+  });
+
+  const renderCellValueHookCasePage = getRenderCellValueHook({
+    scopeId: SourcererScopeName.detections,
+  });
+
+  const sort: AlertsTableConfigurationRegistry['sort'] = [
+    {
+      '@timestamp': {
+        order: 'desc',
+      },
+    },
+  ];
+
+  // register Alert Table on Alert Page
```

Comment:
General remarks, I know we decided to do it like that at first. Looking at it, I am wondering if we will have been easier to pass the `tabId` in the props on the table directly. It is not a deal breaker just wondering what do you think?

Reply:
What do you mean in the props? Actually the hooks that are in registry need that prop ( `TableId` and `ScopeId`) so either we can pass these params to all the hooks in registry props

Alternatively, we can move registry to the props of the table.. which I think will solve this problem. 😉